src/utils/slpReader.ts

```diff
@@ -684,3 +685,49 @@ export function getGameEnd(slpFile: SlpFileType): GameEndType | null {
 
   return gameEndMessage as GameEndType;
 }
+
+export function extractFinalPostFrameUpdates(slpFile: SlpFileType): PostFrameUpdateType[] {
+  const { ref, rawDataPosition, rawDataLength, messageSizes } = slpFile;
+
+  // The following should exist on all replay versions
+  const postFramePayloadSize = messageSizes[Command.POST_FRAME_UPDATE];
+  const gameEndPayloadSize = messageSizes[Command.GAME_END];
+  const frameBookendPayloadSize = messageSizes[Command.FRAME_BOOKEND];
+
+  // Technically this should not be possible
+  if (isUndefined(postFramePayloadSize)) {
```

Comment:
If we don't need to distinguish between null and undefined we can use `exists` function which has already been imported.
```suggestion
  if (exists(postFramePayloadSize)) {
```

Reply:
Why do we have our own exists function?

Reply:
Neither `isNull`, `isNil` or `isUndefined` have been used in the codebase, so by importing and using them, we're introducing another dependency. I'd prefer it if we just used our existing `exists` function. We have our own exists function because it's convenient to handle all the null/undefined cases and it does proper type assertions too.

Reply:
Does the dependency thing matter? It's likely an extremely lightweight function and we're already using lodash in a bunch of places?

In any case, I'll change it to `exists`.

I generally don't like writing our own functions for these kinds of things because we might not consider some edge cases that could cause strange behavior and bugs down the line; I find it better to leave it up to a mature codebase you can be pretty confident will always work right. Also it's an unnecessary function in the codebase you just have to know exists. But regardless, don't feel strongly enough about it to fight for it either.

Reply:
The `exists` function is simple enough to maintain and since we have [tree shaking enabled](https://github.com/project-slippi/slippi-js/pull/89), every additional function we import will increase the bundle size. We might not think this is a big deal but an increase in our bundle size affects and increases the size of every library/project that chooses to import our library.